.github/workflows/pull_request_push_test.yml

```diff
@@ -27,6 +27,12 @@ on:
     # Runs daily at 1 PM UTC (9 PM CST), will send notification to TEAMS_WEBHOOK
     - cron: '00 13 * * *'
 
+# Cancel any in progress workflow from the same PR, branch or tag when a new workflow is triggered
+# Ref: https://stackoverflow.com/questions/66335225/how-to-cancel-previous-runs-in-the-pr-when-you-push-new-commitsupdate-the-curre
+concurrency:
+  group: ${{ github.workflow }}-${{ github.event.pull_request.number || github.ref }}
+  cancel-in-progress: true
+
 jobs:
   gradle_test:
     runs-on: ubuntu-latest
```

Comment:
Shall we also consider 'github.event.pull_request_target.number'?

Reply:
I checked the github webhook event doc https://docs.github.com/en/developers/webhooks-and-events/webhooks/webhook-events-and-payloads, there is not delicated event for `pull_request_target`, so it seems `github.event.pull_request ` should work for pull request from both current repo and forked repo.